catenets/datasets/dataset_acic2016.py

```diff
@@ -0,0 +1,231 @@
+        )
+        X_sq = X[:, NUMERIC_COLS] ** 2
+        mu_0 = mu_0 + np.dot(X_sq, beta_sq).reshape((-1, 1))
+        mu_1 = mu_1 + np.dot(X_sq, beta_sq * omega).reshape((-1, 1))
+
+        if inter:
+            # randomly add some interactions
+            ind_c = np.arange(n_cov)
+            np.random.shuffle(ind_c)
+            inter_list = list()
+            for i in range(0, n_cov - 2, 2):
+                inter_list.append(X[:, ind_c[i]] * X[:, ind_c[i + 1]])
+
+            X_inter = np.array(inter_list).T
+            n_inter = X_inter.shape[1]
+            beta_inter = np.random.choice(
+                coefs_sq, size=n_inter, replace=True, p=[1 - sp_nonlin, sp_nonlin]
+            )
+            omega = np.random.choice(
+                [0, 1], replace=True, size=n_inter, p=[prop_omega, 1 - prop_omega]
+            )
+            mu_0 = mu_0 + np.dot(X_inter, beta_inter).reshape((-1, 1))
+            mu_1 = mu_1 + np.dot(X_inter, beta_inter * omega).reshape((-1, 1))
+
+    ate = np.mean(mu_1 - mu_0)
+    mu_1 = mu_1 - ate + ate_goal
+
+    y = (
+        w * mu_1
+        + (1 - w) * mu_0
+        + np.random.normal(0, error_sd, n_total).reshape((-1, 1))
+    )
+
+    X_train, y_train, w_train, mu_0_train, mu_1_train = (
+        X[ind[: (n_0 + n_1)], :],
+        y[ind[: (n_0 + n_1)]],
+        w[ind[: (n_0 + n_1)]],
+        mu_0[ind[: (n_0 + n_1)]],
+        mu_1[ind[: (n_0 + n_1)]],
+    )
+    X_test, mu_0_t, mu_1_t = (
+        X[ind_test, :],
+        mu_0[ind_test],
+        mu_1[ind_test],
+    )
+
+    return (
+        X_train,
+        w_train,
+        y_train,
+        np.asarray([mu_0_train, mu_1_train]).squeeze().T,
+        X_test,
+        np.asarray([mu_0_t, mu_1_t]).squeeze().T,
+    )
+
+
+def load(
+    data_path: Path,
+    train_ratio: float = 0.8,
+    *args: Any,
+    **kwargs: Any,
+) -> Tuple:
+    """
+    ACIC2016 dataset dataloader.
+        - Download the dataset if needed.
+        - Load the dataset.
+        - Preprocess the data.
+        - Return train/test split.
+
+    Parameters
+    ----------
+    data_path: Path
+        Path to the CSV. If it is missing, it will be downloaded.
+    train_ratio: float
+        Train/test ratio
+
+    Returns
+    -------
+    train_x: array or pd.DataFrame
+        Features in training data.
+    train_t: array or pd.DataFrame
+        Treatments in training data.
+    train_y: array or pd.DataFrame
+        Observed outcomes in training data.
+    train_potential_y: array or pd.DataFrame
+        Potential outcomes in training data.
+    test_x: array or pd.DataFrame
+        Features in testing data.
+    test_potential_y: array or pd.DataFrame
+        Potential outcomes in testing data.
+    """
+    arch = data_path / "data_cf_all.tar.gz"
+
+    download_if_needed(
+        arch, file_id=FILE_ID, unarchive=True, unarchive_folder=data_path
+    )
+
+    csv = data_path / "data_cf_all/x.csv"
```

Comment:
this uses the wrong x_file for this simulation; I think I told you that there's a local preprocessed file (X_trans) that I used for this (which is preprocessed using an old R-script from the original acic2016, that's why it would be easier to just host the pre-transformed file somewhere I think).

Reply:
That being said, there is actually a second set of experiments that I have run on the original acic data + outcomes (so it uses x.csv as well as the other files in the folders in the archive) [these experiments are not in this repo yet]; so it wouldnt hurt to have an option to choose whether to use x_trans.csv or x.csv :)

Reply:
The GDrive share you've sent me doesn't include x_trans.csv, only x.csv.